### ISSUE TITLE
linter: unmatched brackets causes jsdoc matching issues

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#18664]
> I stumbled across a very close variation of this bug today, seems that braces are affected as well if they are not in proper pairs.
> 
> This also leads to a false positive:
> ```
> /**
>  * Escaped characters: )
>  * @returns The escaped text
>  */
> export function escapeCode(text: string): string {
>   return ...
> }
> ```
> 
> Same with `\)` or `())`. Also for other brace types like `]` or `}`.
> While these work: `()`, `\(\)`
>  

 _Originally posted by @eikowagenknecht in [#13776](https://github.com/oxc-project/oxc/issues/13776#issuecomment-3811212279)_

**Comment by Mister-Hope** [comment]
Note: A single starting `(` or `[` breaks this as well. 

<img width="751" height="770" alt="Image" src="https://github.com/user-attachments/assets/c0dc2614-0c22-45ab-ae2e-23723d275410" />

<img width="642" height="756" alt="Image" src="https://github.com/user-attachments/assets/bcf10d2c-8766-4b60-af4c-9fa59ea5661c" />

This happens even in code blocks：

<img width="589" height="315" alt="Image" src="https://github.com/user-attachments/assets/52a3bc2e-5a57-4d0e-9a1b-488712f46bdc" />

````js
/**
 * ```js
 * // Comments with (
 * ```
 *
 * @param searchIndex Search Index
 * @param query Search query
 * @param searchOptions Search options. Each option, if not given, defaults to the corresponding value of `searchOptions` given to the constructor, or to the library default.
 *
 * @returns A sorted array of search results.
 */
````
